jina/hubble/hubio.py

```diff
@@ -0,0 +1,97 @@
+"""Module for wrapping Jina Hub API calls."""
+
+import os
+import argparse
+import json
+from pathlib import Path
+
+from ..helper import (
+    colored,
+    get_full_version,
+)
+from ..logging.logger import JinaLogger
+from ..logging.profile import TimeContext
+from .helper import archive_package
+
+
+HUBBLE_REGISTRY = os.environ.get('HUBBLE_REGISTRY', 'https://hubble.jina.ai')
+
+
+class HubIO:
+    """:class:`HubIO` provides the way to interact with Jina Hub registry.
+    You can use it with CLI to package a directory into a Jina Hub and publish it to the world.
+    Examples:
+        - :command:`jina hub push my_executor/` to push the executor package to Jina Hub
+        - :command:`jina hub pull UUID8` to download the executor identified by UUID8
+    """
+
+    def __init__(self, args: 'argparse.Namespace'):
+        """Create a new HubIO.
+        :param args: arguments
+        """
+        self.logger = JinaLogger(self.__class__.__name__, **vars(args))
+        self.args = args
+
+    def push(self) -> None:
+        """Push the executor pacakge to Jina Hub."""
+
+        import requests
+        import hashlib
+
+        is_public = True
+        if self.args.private:
+            is_public = False
+            self.args.public = False
```

Comment:
this looks redundant, we only need `--public` we don't need `--private`,

when not given `--public` (by default), then `self.args.public` will be False, so it is private.

so default is `private`

Reply:
done

Reply:
@hanxiao I think we should keep the `--public` and `--private`. We may use them with the `--force` option.

We people do `jina hub push --public --force UUID8 --secret TOKEN .`. They may want to change the visibility of this executor. From private to public, or from public to private.

Reply:
And why the default is `private`? As far as I can see, if we want more public shared executors in jina hub, public default value could be better.